pkg/shp/cmd/build/upload.go

```diff
@@ -226,13 +272,34 @@ func (u *UploadCommand) Run(p *params.Params, ioStreams *genericclioptions.IOStr
 
 	if u.follow {
 		// when follow flag is enabled, instantiating the "follower" to live tail logs
-		if u.follower, err = p.NewFollower(u.Cmd().Context(), *br, ioStreams); err != nil {
+		if u.follower, err = p.NewFollower(u.Cmd().Context(), types.NamespacedName{Namespace: br.Namespace, Name: br.Name}, ioStreams); err != nil {
 			return err
 		}
 	}
 
-	// registering the routine that will react upon build pod state changes
-	u.pw.WithOnPodModifiedFn(u.onPodModifiedEvent)
+	switch u.method {
+	case streaming:
+		// registering the routine that will react upon build pod state changes
+		u.pw.WithOnPodModifiedFn(u.onPodModifiedEvent)
+
+	case bundling:
+		_, err = bundle.Push(u.cmd.Context(), ioStreams, u.sourceDir, u.sourceBundleImage)
+		if err != nil {
+			return err
+		}
+
+		u.pw.WithOnPodModifiedFn(func(pod *corev1.Pod) error {
```

Comment:
Please isolate this logic in the same way `u.onPodModifiedEvent` does.

Reply:
Out of curiosity, why would you prefer this over the ad-hoc function? The inline ad-hoc function makes sense here, as it keeps the content in the exact spot where it is needed, this function is only used once (no need for reusability), and without a separate function I can directly access the variables required (no need to make it referenced).